`HttpClient.Timeout` when retrieving bot profile crashes the app

Logs:

```log
fail: LichessChallenger.Worker[0]
      Error querying online status of Lynx_BOT: The request was canceled due to the configured HttpClient.Timeout of 100 seconds elapsing.
05:26:18 | [ERROR] Error querying online status of Lynx_BOT:    The request was canceled due to the configured HttpClient.Timeout of 100 seconds elapsing.
fail: Microsoft.Extensions.Hosting.Internal.Host[9]
      BackgroundService failed
      System.NullReferenceException: Can't find Lichess username Lynx_BOT
         at LichessChallenger.Worker.ExecuteAsync(CancellationToken stoppingToken) in LichessChallenger.dll:token 0x600000b+0x4df
         at Microsoft.Extensions.Hosting.Internal.Host.TryExecuteBackgroundServiceAsync(BackgroundService ) in Microsoft.Extensions.Hosting.dll:token 0x6000060+0x76
05:26:18 | [ERROR] BackgroundService failed System.NullReferenceException: Can't find Lichess username Lynx_BOT
   at LichessChallenger.Worker.ExecuteAsync(CancellationToken stoppingToken) in LichessChallenger.dll:token 0x600000b+0x4df
   at Microsoft.Extensions.Hosting.Internal.Host.TryExecuteBackgroundServiceAsync(BackgroundService ) in Microsoft.Extensions.Hosting.dll:token 0x6000060+0x76
crit: Microsoft.Extensions.Hosting.Internal.Host[10]
      The HostOptions.BackgroundServiceExceptionBehavior is configured to StopHost. A BackgroundService has thrown an unhandled exception, and the IHost instance is stopping. To avoid this behavior, configure this to Ignore; however the BackgroundService will not be restarted.
      System.NullReferenceException: Can't find Lichess username Lynx_BOT
         at LichessChallenger.Worker.ExecuteAsync(CancellationToken stoppingToken) in LichessChallenger.dll:token 0x600000b+0x4df
         at Microsoft.Extensions.Hosting.Internal.Host.TryExecuteBackgroundServiceAsync(BackgroundService ) in Microsoft.Extensions.Hosting.dll:token 0x6000060+0x76
05:26:18 | [FATAL] The HostOptions.BackgroundServiceExceptionBehavior is configured to StopHost. A BackgroundService has thrown an unhandled exception, and the IHost instance is stopping. To avoid this behavior, configure this to Ignore; however the BackgroundService will not be restarted. System.NullReferenceException: Can't find Lichess username Lynx_BOT
   at LichessChallenger.Worker.ExecuteAsync(CancellationToken stoppingToken) in LichessChallenger.dll:token 0x600000b+0x4df
   at Microsoft.Extensions.Hosting.Internal.Host.TryExecuteBackgroundServiceAsync(BackgroundService ) in Microsoft.Extensions.Hosting.dll:token 0x6000060+0x76
05:26:18 | [INFO] Application is shutting down...
```

--- a/src/LichessChallenger/Worker.cs
+++ b/src/LichessChallenger/Worker.cs
@@ -63,8 +63,8 @@ namespace LichessChallenger
                 await Task.Delay(_timeBetweenChallenges, stoppingToken);
 
                 // Seems repetitive, but we need to check IsPlaying status every time
-                User me = await FindUser(_me, stoppingToken) ?? throw new NullReferenceException($"Can't find Lichess username {_me}");
-                if (!me.IsOnline || me.IsPlaying)
+                User? me = await FindUser(_me, stoppingToken);
+                if (me is null || !me.IsOnline || me.IsPlaying)
                 {
                     continue;
                 }
@@ -99,7 +99,7 @@ namespace LichessChallenger
                 }
                 catch (Exception e)
                 {
-                    _logger.LogError($"Error challenging {rivalUsername}:\t{e.Message}");
+                    _logger.LogError(e, $"Error challenging {rivalUsername}");
                     continue;
                 }
 
@@ -118,7 +118,7 @@ namespace LichessChallenger
             }
             catch (Exception e) // HttpRequestException, NotSupportedException or JsonException
             {
-                _logger.LogError($"Error querying online status of {username}:\t{e.Message}");
+                _logger.LogError(e, $"Error querying online status of {username}");
             }
 
             return user;

